datastore/api/snippets.py

```diff
@@ -0,0 +1,820 @@
+        'type': 'Work',
+        'done': False,
+        'priority': 8,
+        'description': 'Integrate Cloud Datastore'
+    })
+
+    client.put_multi([task1, task2])
+    # [END batch_upsert]
+
+    return task1, task2
+
+
+def batch_lookup(client):
+    # Create the entities we will lookup.
+    batch_upsert(client)
+
+    # [START batch_lookup]
+    keys = [
+        client.key('Task', 1),
+        client.key('Task', 2)
+    ]
+
+    tasks = client.get_multi(keys)
+    # [<Entity Task 1>, <Entity Task 2>]
+    # [END batch_lookup]
+
+    return tasks
+
+
+def batch_delete(client):
+    # Create the entities we will delete.
+    batch_upsert(client)
+
+    # [START batch_delete]
+    keys = [
+        client.key('Task', 1),
+        client.key('Task', 2)
+    ]
+
+    client.delete_multi(keys)
+    # [END batch_delete]
+
+    return keys
+
+
+def unindexed_property_query(client):
+    # Create the entity that we're going to query.
+    upsert(client)
+
+    # [START unindexed_property_query]
+    query = client.query(kind='Task')
+    query.add_filter('description', '=', 'Learn Cloud Datastore')
+    tasks = list(query.fetch())
+    # [<Entity Task>, ... ]
+    # [END unindexed_property_query]
+
+    return tasks
+
+
+def basic_query(client):
+    # Create the entity that we're going to query.
+    upsert(client)
+
+    # [START basic_query]
+    query = client.query(kind='Task')
+    query.add_filter('done', '=', False)
+    query.add_filter('priority', '>=', 4)
+    query.order = ['-priority']
+
+    tasks = list(query.fetch())
+    # [END basic_query]
+
+    return tasks
+
+
+def projection_query(client):
+    # Create the entity that we're going to query.
+    task = datastore.Entity(client.key('Task'))
+    task.update({
+        'type': 'Personal',
+        'done': False,
+        'priority': 4,
+        'description': 'Learn Cloud Datastore',
+        'percent_complete': 0.5
+    })
+    client.put(task)
+
+    # [START projection_query]
+    query = client.query(kind='Task')
+    query.projection = ['priority', 'percent_complete']
+    # [END projection_query]
+
+    # [START run_query_projection]
+    priorities = []
+    percent_completes = []
+
+    for task in query.fetch():
+        priorities.append(task['priority'])
+        percent_completes.append(task['priority'])
+    # [END run_query_projection]
+
+    return priorities, percent_completes
+
+
+def ancestor_query(client):
+    task = datastore.Entity(
+        client.key('TaskList', 'default', 'Task'))
+    task.update({
+        'type': 'Personal',
+        'description': 'Learn Cloud Datastore',
+    })
+    client.put(task)
+
+    # [START ancestor_query]
+    ancestor = client.key('TaskList', 'default')
+    query = client.query(kind='Task', ancestor=ancestor)
+    tasks = list(query.fetch())
+    # [END ancestor_query]
+
+    return tasks
+
+
+def run_query(client):
+
+    # [START run_query]
+    query = client.query()
+    results = list(query.fetch())
```

Comment:
Commented a couple of times, but *most* of the query snippets should only define the query, not run it.